libraries/AC_Autorotation/AC_Autorotation.cpp

```diff
@@ -0,0 +1,237 @@
+
+    // @Param: HS_SET_PT
+    // @DisplayName: Target Head Speed
+    // @Description: The target head speed in RPM during autorotation.  Start by setting to desired hover speed and tune from there as neccassary.
+    // @Units: RPM
+    // @Range: 1000 2800
+    // @Increment: 1
+    // @User: Advanced
+    AP_GROUPINFO("HS_SET_PT", 2, AC_Autorotation, _param_head_speed_set_point, 1500.0f),
+
+    // @Param: TARG_SP
+    // @DisplayName: Target Glide Ground Speed
+    // @Description: Target ground speed in cm/s for the autorotation controller to try and achieve/ maintain during the glide phase.
+    // @Units: cm/s
+    // @Range: 800 2000
+    // @Increment: 50
+    // @User: Advanced
+    AP_GROUPINFO("TARG_SP", 3, AC_Autorotation, _param_target_speed, SPD_HGT_CONTROLLER_GND_SPEED_TARGET),
+
+    // @Param: COL_FILT_E
+    // @DisplayName: Entry Phase Collective Filter
+    // @Description: Cut-off frequency for collective low pass filter.  For the entry phase.  Acts as a following trim.  Must be higher than COL_FILT_G.
+    // @Units: Hz
+    // @Range: 0.2 0.5
+    // @Increment: 0.01
+    // @User: Advanced
+    AP_GROUPINFO("COL_FILT_E", 4, AC_Autorotation, _param_col_entry_cutoff_freq, HS_CONTROLLER_ENTRY_COL_FLITER),
+
+    // @Param: COL_FILT_G
+    // @DisplayName: Glide Phase Collective Filter
+    // @Description: Cut-off frequency for collective low pass filter.  For the glide phase.  Acts as a following trim.  Must be lower than COL_FILT_E.
+    // @Units: Hz
+    // @Range: 0.03 0.15
+    // @Increment: 0.01
+    // @User: Advanced
+    AP_GROUPINFO("COL_FILT_G", 5, AC_Autorotation, _param_col_glide_cutoff_freq, HS_CONTROLLER_GLIDE_COL_FLITER),
+
+    // @Param: AS_ACC_MAX
+    // @DisplayName: Forward Acceleration Limit
+    // @Description: Maximum forward acceleration to apply in speed controller.
+    // @Units: cm/s/s
+    // @Range: 50 200
+    // @Increment: 10
+    // @User: Advanced
+    AP_GROUPINFO("AS_ACC_MAX", 6, AC_Autorotation, _param_accel_max, SPD_HGT_CONTROLLER_MAX_ACCEL),
+
+    // @Param: BAIL_TIME
+    // @DisplayName: Bail Out Timer
+    // @Description: Time in seconds from bail out initiated to the exit of autorotation flight mode.
+    // @Units: s
+    // @Range: 0.5 4
+    // @Increment: 0.1
+    // @User: Advanced
+    AP_GROUPINFO("BAIL_TIME", 7, AC_Autorotation, _param_bail_time, AROT_BAIL_OUT_TIME),
+
+    // @Param: HS_SENSOR
+    // @DisplayName: Main Rotor RPM Sensor 
+    // @Description: Allocate the RPM sensor instance to use for measuring head speed.  RPM1 = 0.  RPM2 = 1.
+    // @Units: s
+    // @Range: 0.5 3
+    // @Increment: 0.1
+    // @User: Advanced
+    AP_GROUPINFO("HS_SENSOR", 8, AC_Autorotation, _param_rpm_instance, 0),
+
+    AP_GROUPEND
+};
+
+
+//initialisation of head speed controller
+void AC_Autorotation::init_hs_controller()
+{
+    //set initial collective position to be the collective position on initialisation
+    _collective_out = 0.4f;//_motors.get_throttle();
```

Comment:
let's remove the //_motors.get_throttle() part.

Reply:
done